src/components/Thought.js

```diff
@@ -246,6 +246,7 @@ export const Thought = connect(({ cursor, cursorBeforeEdit, expanded, expandedCo
     // this is a bit of a hack since the bullet transform checks leaf instead of expanded
     leaf: children.length === 0 || (isEditing && globals.suppressExpansion),
     'has-only-child': children.length === 1,
+    hidden: isPaginated,
```

Comment:
Instead of hiding paged thoughts, they should not be rendered. Part of the reason for pagination is for performance. Rendering and hiding them with CSS somewhat defeats this purpose. I recommend using a different approach that ensures that the correct thoughts are rendered or re-rendered in React based on the paging logic.

Reply:
Then cursorNext and cursorDown will not be supported. as they are working based on DOM operations.

Reply:
I have to do pagination like this as cursorNext and cursorDown are working by DOM (which is not a good pattern in React application, but we have what we have.

Reply:
I agree that the DOM manipulation from `selectNextEditable` is undesirable. Hopefully we can refactor that soon.

However I think this problem is solvable without using the DOM. If the `cursor` is on the last thought and there are more pages, then we know that the next thought is `getThoughtAfter(cursor)`. Then `setCursor` can be dispatched with the next thought.

Reply:
Fixed this